pkg/binding/bindingpolicy-resolution.go

```diff
@@ -148,21 +177,23 @@ func (resolution *bindingPolicyResolution) matchesBindingSpec(bindingSpec *v1alp
 	}
 
 	// check workload
-	if len(resolution.objectIdentifierToResourceVersion) != len(bindingSpec.Workload.ClusterScope)+
+	if len(resolution.objectIdentifierToData) != len(bindingSpec.Workload.ClusterScope)+
 		len(bindingSpec.Workload.NamespaceScope) {
 		return false
 	}
 
-	objectRefSetFromWorkload := bindingObjectRefAndVersionSetFromWorkload(&bindingSpec.Workload)
+	objRefAndDataSetFromWorkload := bindingObjectRefAndDataSetFromWorkload(&bindingSpec.Workload)
 
-	for objIdentifier, objResourceVersion := range resolution.objectIdentifierToResourceVersion {
-		if !objectRefSetFromWorkload.Has(objectRefAndVersion{
+	for objIdentifier, objData := range resolution.objectIdentifierToData {
+		// check if object ref exists, then check if the object data matches
+		if !objRefAndDataSetFromWorkload.Has(objectRefAndData{
```

Comment:
This is going to compare `objectRefAndData` according to Go `==` --- which is not what you want, because that will compare pointer values for the `*objectData` field rather than doing a more semantic equality test.

Reply:
Apologies, thought I fixed that.